src/core/tsi/s2a/handshaker/s2a_handshaker_client.cc

```diff
@@ -86,11 +86,14 @@ tsi_result S2AHandshakerClient::ClientStart() {
   }
   grpc_byte_buffer_destroy(send_buffer_);
   send_buffer_ = buffer;
-  tsi_result call_result = MakeGrpcCall(/*is_start=*/true);
-  if (call_result != TSI_OK) {
-    gpr_log(GPR_ERROR, kS2AMakeGrpcCallFailed);
+  if (!no_calls_for_testing_) {
+    tsi_result call_result = MakeGrpcCall(/* is_start=*/true);
+    if (call_result != TSI_OK) {
+      gpr_log(GPR_ERROR, kS2AMakeGrpcCallFailed);
+      return call_result;
+    }
   }
-  return call_result;
+  return TSI_OK;
```

Comment:
Is it possible to refactor the code from lines 89 to 94? It seems to be used repeatedly in three different places.

Reply:
Done in next commit.